File: app/src/main/java/com/haoyu/app/fragment/PictureViewerFragment.java
package com.haoyu.app.fragment;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;

import com.bumptech.glide.Glide;
import com.haoyu.app.lego.student.R;
import com.haoyu.app.view.TouchImageView;

/**
 * 创建日期：2017/11/16.
 * 描述:图片预览
 * 作者:xiaoma
 */

public class PictureViewerFragment extends Fragment {

    private Context context;
    private String url;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.context = context;
        Bundle bundle = getArguments();
        url = bundle.getString("url");
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        TouchImageView imageView = new TouchImageView(context);
        imageView.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
        imageView.setBackgroundColor(ContextCompat.getColor(context, R.color.black));
        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT);
        params.gravity = Gravity.CENTER;
        imageView.setLayoutParams(params);
        loadImage(imageView);
        return imageView;
    }

    private void loadImage(TouchImageView imageView) {
        Glide.with(context).load(url).placeholder(R.drawable.ic_placeholder).fitCenter().dontAnimate().into(imageView);
    }
}

File: app/src/main/java/com/haoyu/app/fragment/PictureViewerFragment.kt
package com.haoyu.app.fragment

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import com.bumptech.glide.Glide
import com.haoyu.app.lego.student.R
import com.haoyu.app.view.TouchImageView

/**
 * 创建日期：2018/1/17.
 * 描述:图片预览
 * 作者:xiaoma
 */
class PictureViewerFragment : Fragment() {
    private lateinit var activity: Context
    private var url: String? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        activity = context
        url = arguments?.getString("url")
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val iv = TouchImageView(context)
        iv.scaleType = ImageView.ScaleType.CENTER_INSIDE
        iv.setBackgroundColor(ContextCompat.getColor(activity, R.color.black))
        val params = LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
        params.gravity = Gravity.CENTER
        iv.layoutParams = params
        loadImage(iv)
        return iv
    }

    private fun loadImage(iv: TouchImageView) {
        Glide.with(activity).load(url).placeholder(R.drawable.ic_placeholder).fitCenter().dontAnimate().into(iv)
    }
}